Song title stays unset unless explicitly set by user

dlbeswick opened the pull request:
The song filename will be shown in the window title bar if a song title hasn't been set.

The existing behaviour is to set the song title field to the file's basename when the song is saved.

I tend to start a song, play around with a few things, and then save different versions as I go. If the song title is set to the filename on first save, then I get many different song files all having the same song title in the titlebar, which confuses me. So, my preference is to show the filename in the title bar until the song has a title explicitly set.

ensonic commented on src/lib/core/song-info.c:
```diff
@@ -331,6 +331,19 @@ bt_song_info_tick_to_m_s_ms (const BtSongInfo * const self, const gulong tick,
   bt_song_info_time_to_m_s_ms (self, ts, m, s, ms);
 }
 
+gchar *
```
Please add a docstring and add the function name to the docs (https://github.com/Buzztrax/buzztrax/blob/master/docs/reference/bt-core/buzztrax-core-sections.txt#L440).
Alternative would be to keep the function private and expose this through a new property on the class.

dlbeswick replied:
Thanks for picking up on the missing docstring. I've added one and updated the sections file.

It is useful to know that I can use a readonly property to calculate these sort of derived values, I'll keep that in mind in future.

What's the purpose of buzztrax-core-sections.txt?

ensonic replied:
*-sections.txt maps symbols that should be exposed to the docs to the right section. This can be also auto-generated, but then some cornercase cannot be addressed.

ensonic commented:
Thanks! The logic makes sense, could you try adding a test for this as well?

dlbeswick commented:
Sure! I'll add a test.

dlbeswick commented:
Needs some tidying up when the untitled song is saved. Default filename presented to the user is ".xml"

ensonic commented:
I'll wait for a test part of the PR before merging. Should be relative straight forward here.

dlbeswick commented:
Ok, I've fixed the above extra issue and have added tests for this case. I'll create a separate pull requests to test the other functionality that you recommended.

The original change was not very good anyway; it only updated the title when the song status changed from saved to unsaved. I've fixed that.

There were some issues that were causing "make check" to fail for my out-of-tree build that I've addressed; hope that's all good.

dlbeswick commented on docs/help/bt-edit/Makefile.am:
```diff
@@ -10,24 +10,29 @@ HELP_MEDIA += \
 	version.entities
```
This file was re-sorted by "aspell".

ensonic replied:
Thanks!

dlbeswick commented on src/ui/edit/main-window.c:
```diff
@@ -272,6 +273,44 @@ on_song_unsaved_changed (const GObject * object, GParamSpec * arg,
+  BtSongInfo* song_info;
+  g_object_get (song, "song-info", &song_info, NULL);
+
+  g_signal_connect (song_info, "notify::name",
```
I presume it's ok to do this without disconnecting the old handler because the old handler will automatically be disconnected when the old "song-info" object is finalised. Is that right?

dlbeswick commented on configure.ac:
```diff
@@ -672,7 +672,6 @@ AC_SUBST(ARCH_CCASFLAGS)
 AC_CONFIG_FILES(Makefile \
   docs/version.entities \
   docs/help/bt-edit/Makefile \
-  docs/help/bt-edit/C/Makefile \
```
I've folded the tasks in "bt-edit/C/Makefile" into "bt-edit/Makefile" to deal with out-of-tree build issue I had. There's a comment in "bt-edit/Makefile" explaining why.
